### ISSUE TITLE
Swap nodes

### ISSUE THREAD

**Issue by calebeby** [aaronik/treewalker.nvim#4]
Hey, thanks for this plugin! I'm coming from https://github.com/ziontee113/syntax-tree-surfer which is unmaintained. This is the exact replacement I was looking for!

It would be amazing if your plugin would have the ability to move/swap nodes on the same level like syntax-tree-surfer could. For example, to swap list items in markdown files, even if the list items are more than one line long and have nested content.

Totally ok if you aren't interested in this feature but I thought I'd put it out there.

Thanks again for this really awesome plugin!

**Comment by aaronik** [comment]
Oh wow I sure am thinking about this a lot now! I mean, this thing does identify the nodes, it wouldn't be hard at all to swap them..

**Comment by aaronik** [comment]
Closing as implemented in d09fbe7b10d39191ca4a62213517a758465d1b38 :)

**Comment by ccjmne** [comment]
Wow, this looks good, @aaronik!

I checked it out, tried it out, and it works well.
As you'll certainly infer by the tangent that follows though, I won't be using it much right away, I have a new need to address!

I'll take this opportunity to mention that I've kept up with checking out your humourous commits messages and I do read your code with great care (it's my first time reading through some Lua code with any sort of seriousness) and it's quite fun ;)

`------------------------ >8 ------------------------`

Something somewhere in my set-up doesn't quite support passing a physical `<C-S-j>` from my keyboard all the way down to `Neovim` through `tmux`, and for some reason, I'd never noticed it before? Anyhow, it's not `tmux`, and it's most definitely not `Neovim`... so, the terminal emulator?

I've been around the Internet a bit so I know all the cool people are running some sort of funky stuff there, and I managed to systematically go: _"this is not a rabbit hole I need to explore right now"_... Has the time come?

Just because _you_ had the audacity to showcase your mappings and use that! How shall I justify to my family on Christmas day tomorrow that I _need_ rather to check out all the terminal emulators in existence? Will you write me some sort of a doctor's note? ;)

Cheers!
